File: server/feed5/swagger_server/daos/inbound_dao.py
from ..models.inbound import Inbound as InboundModel
from .cassandra_dao import CassandraDAO

class Inbound(CassandraDAO):
    _from = None
    to = None
    subject = None
    story = None
    insert = 'insert_inbound'
    query = 'select_inbound'

    def __init__(self, to: int, _from: int = None, subject: str = None, story: str = None):
        self._from = _from
        self.to = to
        self.subject = subject
        self.story = story
        self.prepare(self.insert, 'insert into Inbound (ParticipantID, FromParticipantID, Occurred, Subject, Story) values (?, ?, now(), ?, ?)')
        self.prepare(self.query, 'select toTimestamp(occurred) as occurred, fromparticipantid, subject, story from Inbound where participantid = ? order by occurred desc')

    def makeInbound(self, dikt):
        retVal = {}
        retVal['occurred'] = dikt['occurred'].strftime('%Y-%m-%d')
        retVal['from'] = dikt['fromparticipantid']
        retVal['to'] = self.to
        retVal['subject'] = dikt['subject']
        retVal['story'] = dikt['story']
        return retVal

    def save(self):
        self.execute(self.insert, [ self.to, self._from, self.subject, self.story ])

    def load(self):
        return self.execute(self.query, [ self.to ], self.makeInbound)

    def __repr__(self):
        return '{"from":%d, "to":%d, "subject":"%s", "story":"%s"}' % (self._from, self.to, self.subject, self.story)

    def _from(self) -> int:
        return self._from

    def to(self) -> int:
        return self.to

    def subject(self) -> str:
        return self.subject

    def story(self) -> str:
        return self.story

File: server/feed11/src/main/scala/info/glennengstrand/news/dao/InboundDao.scala
package info.glennengstrand.news.dao

import scala.concurrent.Future
import scala.concurrent._
import ExecutionContext.Implicits.global
import scala.collection.JavaConverters._

import com.datastax.oss.driver.api.core.cql.{BoundStatement, PreparedStatement}
import info.glennengstrand.news.model.Inbound

class InboundDao extends DataAccess[Inbound] {
  private val selectCql = "select toTimestamp(occurred) as occurred, fromparticipantid, subject, story from Inbound where participantid = ? order by occurred desc"
  private val insertCql = "insert into Inbound (ParticipantID, FromParticipantID, Occurred, Subject, Story) values (?, ?, now(), ?, ?)"
  private lazy val insertStmt = CassandraDao.session.prepare(insertCql)
  private lazy val selectStmt = CassandraDao.session.prepare(selectCql)
  
  override def fetchSingle(id: Int): Future[Inbound] = {
    Future(Inbound(None, None, None, None, None))
  }
  override def insert(ib: Inbound): Future[Inbound] = {
    val bs = insertStmt.bind(new java.lang.Integer(extractId(ib.to.get.toString.asInstanceOf[String]).toInt), new java.lang.Integer(extractId(ib.from.get.toString.asInstanceOf[String]).toInt), ib.subject.get.toString.asInstanceOf[String], ib.story.get.toString.asInstanceOf[String])
    CassandraDao.session.execute(bs)
    Future(ib)
  }
  override def fetchMulti(id: Int): Future[Seq[Inbound]] = {
    val bs = selectStmt.bind(id.asInstanceOf[Object])
    val retVal = for {
      r <- CassandraDao.session.execute(bs).iterator().asScala
    } yield Inbound(Option(toLink(r.getInt(1).toLong)), Option(toLink(id)), Option(r.getInstant(0).toString()), Option(r.getString(2)), Option(r.getString(3)))
    Future(retVal.toSeq)
  }
}